src/network/mcpe/NetworkSession.php

```diff
@@ -419,10 +419,31 @@ public function handleEncoded(string $payload) : void{
 	 * @throws PacketHandlingException
 	 */
 	public function handleDataPacket(Packet $packet, string $buffer) : void{
+
+		$playerName = $this->getPlayer()?->getName();
+		if (!isset($this->packetsReceived[$playerName])) {
+			$this->packetsReceived[$playerName] = [
+				'count' => 0,
+				'time' => time(),
+			];
+		}
+
+		$playerPackets = &$this->packetsReceived[$playerName];
+		$playerPackets['count']++;
 		if(!($packet instanceof ServerboundPacket)){
 			throw new PacketHandlingException("Unexpected non-serverbound packet");
 		}
 
+		if (strlen($buffer) > 16500 && $packet->getName() === "InventoryTransactionPacket") {
+			$this->logger->debug("Huge InventoryTransactionPacket: " . base64_encode($buffer));
+			throw new PacketHandlingException("InventoryTransactionPacket too big");
+		}
+		if (strlen($buffer) > 31968) {
+			if (time() - $playerPackets['time'] >= 2 && $playerPackets['count'] > 5) {
+				$this->getPlayer()?->kick("Too fast packet");
```

Comment:
Here there is no interruption
```suggestion
				throw new PacketHandlingException("Too fast packet");
```
PacketHandlingException hides the reason and may be difficult to identify
(Server administrators get confused by the display of "Bad packet (error ID $errorId):")